### PR TITLE
Fix bi_valid handling in deflate_quick()

### PR DESCRIPTION
**Pull request by iii-i**
The attached test started failing after commit ad89d5131b29 ("Don't
write end of last block when returning finish_started."): either with
"bi_buf not flushed" message in debug builds, or by producing corrupted
output in release builds.
    
The problem is that we must not return finish_started when bi_buf is
not empty, because the bits there will be lost. Fix by checking that
bi_valid is not 0.

### CODE REVIEW

File: test/deflate_quick_bi_valid.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    PREFIX3(stream) strm;
+    memset(&strm, 0, sizeof(strm));
+
+    int ret = PREFIX(deflateInit2)(&strm, 1, Z_DEFLATED, 31, 1, Z_FILTERED);
+    if (ret != Z_OK) {
+        fprintf(stderr, "deflateInit2() failed with code %d\n", ret);
+        return EXIT_FAILURE;
+    }
+
+    z_const Bytef next_in[554] =
```
Comment (mtl1979) [inline test/deflate_quick_bi_valid.c]: New code should not use ```Bytef```... It's deprecated in zlib-ng.

Comment (iii-i) [reply test/deflate_quick_bi_valid.c]: Will change to `unsigned char`.

**Comment by Dead2** [comment]
@iii-i FYI, the `macOS GCC` test is randomly failing, that is a known problem and not your fault.

**Comment by nmoinvaz** [comment]
The changes look pretty straight forward. Having a test case helps validate the edge case. Based on the description of `finish_started` it makes sense that it should need to check `bi_valid`. Nice work tracking this down!

image

**Comment by iii-i** [comment]
Thanks for reviewing! I've rebased and pushed the `unsigned char` change.

**Comment by Dead2** [comment]
Github is struggling today, so CI is not running (scarily it says "All checks have passed" as if nothing is wrong).
Hopefully they get it fixed shortly.

**Comment by Dead2** [comment]
GHA is up again, but the missing checks are still missing, and Github pretends that everything is fine..  Can't even re-run them without a forced re-push.

**Comment by Dead2** [comment]
@iii-i I pushed a dummy commit so the CI would run. 😄 
Please do a force push without it whenever you get back online.

**Comment by Dead2** [comment]
Thanks! 😄
